File: RepoTransBench/source_projects/Java/google_mail-importer/src/test/java/to/lean/tools/gmail/importer/FlagsModuleTest.java
package to.lean.tools.gmail.importer;

import com.google.inject.Guice;
import com.google.inject.Injector;
import org.junit.Test;
import static org.junit.Assert.*;

public class FlagsModuleTest {
    @Test
    public void testFlagsModuleBindsArguments() {
        CommandLineArguments args = new CommandLineArguments();
        args.mailboxFileName = "foo";
        Injector injector = Guice.createInjector(new FlagsModule(args));
        CommandLineArguments injected = injector.getInstance(CommandLineArguments.class);
        assertEquals("foo", injected.mailboxFileName);
        // They should be the same instance
        assertSame(args, injected);
    }
}

File: RepoTransBench/target_projects/Java/Go/google_mail-importer/tests/original/flags_module_test.go
package original

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type FlagsModule struct {
	Args *CommandLineArguments
}

func NewFlagsModule(args *CommandLineArguments) *FlagsModule {
	return &FlagsModule{Args: args}
}

func (fm *FlagsModule) Inject() *CommandLineArguments {
	return fm.Args
}

func TestFlagsModuleBindsArguments(t *testing.T) {
	args := defaultCommandLineArguments()
	mailbox := "foo"
	args.MailboxFileName = &mailbox
	module := NewFlagsModule(&args)
	injected := module.Inject()
	assert.Equal(t, "foo", *injected.MailboxFileName)
	assert.Equal(t, &args, injected)
}